src/inspector_profiler.cc

```diff
@@ -180,6 +184,117 @@ void V8CoverageConnection::End() {
   DispatchMessage(end);
 }
 
+void V8CpuProfilerConnection::OnMessage(
+    const v8_inspector::StringView& message) {
+  Debug(env(),
+        DebugCategory::INSPECTOR_PROFILER,
+        "Receive cpu profiling message, ending = %s\n",
+        ending_ ? "true" : "false");
+  if (!ending_) {
+    return;
+  }
+  Isolate* isolate = env()->isolate();
+  Local<Context> context = env()->context();
+  HandleScope handle_scope(isolate);
```

Comment:
Strictly speaking, these two lines should be reversed – it’s only because of an implementation detail that `Environment::context()` doesn’t actually allocate a handle itself